terraform/variables.tf

```
@@ -0,0 +1,42 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+variable "app_name" {
+  description = "Name of the application in the Juju model."
+  type        = string
+  default     = "cinder-csi"
+}
+
+variable "base" {
+  description = "Ubuntu bases to deploy the charm onto"
+  type        = string
+  default     = "ubuntu@22.04"
+
+  validation {
+    condition     = contains(["ubuntu@20.04", "ubuntu@22.04"], var.base)
```

Maybe we can say something like this?
```suggestion
    condition     = var.base == "ubuntu@20.04" || var.base == "ubuntu@22.04"
```

---

I think the `contains` function works better here: https://developer.hashicorp.com/terraform/language/expressions/custom-conditions#contains-function

---

and if we get the chance, we'll add `ubuntu@24.04`.  Are you ok with this @HomayoonAlimohammadi